rest_api/sawtooth_rest_api/exceptions.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from aiohttp import web
+from sawtooth_rest_api.protobuf import client_pb2 as client
+
+
+class WrongBodyType(web.HTTPBadRequest):
+    def __init__(self):
+        message = 'Expected an octet-stream encoded Protobuf binary'
+        super().__init__(reason=message)
+
+
+class ValidatorUnavailable(web.HTTPServiceUnavailable):
+    def __init__(self):
+        message = 'Could not reach validator, validator timed out'
+        super().__init__(reason=message)
+
+
+class _ErrorTrap(object):
```

Comment:
I don't understand the exception strategy; could probably use more documentation here describing the overall approach.